shared/actions/login.js

```diff
@@ -3,88 +3,48 @@
-import HiddenString from '../util/hidden-string'
 import openURL from '../util/open-url'
-import {RPCError} from '../util/errors'
-import {chatTab, loginTab, peopleTab, isValidInitialTab} from '../constants/tabs'
-import {deletePushTokenSaga} from './push'
 import {getExtendedStatus} from './config'
 import {isMobile} from '../constants/platform'
 import appRouteTree from '../app/routes-app'
 import loginRouteTree from '../app/routes-login'
-import {pathSelector, navigateTo, navigateAppend, switchRouteDef} from './route-tree'
 import {type InitialState} from '../constants/types/config'
 import {type TypedState} from '../constants/reducer'
+import {niceError} from '../util/errors'
+import HiddenString from '../util/hidden-string'
 
 // Login dips into the routing dep tree, so we need to tell
 // webpack that we can still handle updates that propagate to here.
 export function setupLoginHMR(cb: () => void) {
   module.hot && module.hot.accept(['../app/routes-app', '../app/routes-login'], cb)
 }
 
-const deviceType: DevicesTypes.DeviceType = isMobile ? 'mobile' : 'desktop'
-const InputCancelError = {
-  code: RPCTypes.constantsStatusCode.scinputcanceled,
-  desc: 'Cancel Login',
-}
-
-function _generateQRCode(_, state: TypedState) {
-  if (state.login.codePageTextCode) {
-    return Saga.put(
-      LoginGen.createSetQRCode({
-        codePageQrCode: new HiddenString(Constants.qrGenerate(state.login.codePageTextCode.stringValue())),
-      })
-    )
-  }
-}
-
 function* getAccounts(): Generator<any, void, any> {
   try {
-    yield Saga.put(WaitingGen.createIncrementWaiting({key: DevicesConstants.waitingKey}))
-    const accounts = yield Saga.call(RPCTypes.loginGetConfiguredAccountsRpcPromise)
+    const accounts = yield Saga.call(
+      RPCTypes.loginGetConfiguredAccountsRpcPromise,
+      undefined,
+      DevicesConstants.waitingKey
+    )
     yield Saga.put(LoginGen.createConfiguredAccounts({accounts}))
   } catch (error) {
     yield Saga.put(LoginGen.createConfiguredAccountsError({error}))
-  } finally {
-    yield Saga.put(WaitingGen.createDecrementWaiting({key: DevicesConstants.waitingKey}))
-  }
-}
-
-function* setCodePageOtherDeviceRole(codePageOtherDeviceRole: Types.DeviceRole): Generator<any, void, any> {
-  const state: TypedState = yield Saga.select()
-  if (state.login.codePageMyDeviceRole == null) {
-    logger.warn("my device role is null, can't setCodePageOtherDeviceRole. Bailing")
-    return
-  }
-
-  const codePageMode = Constants.defaultModeForDeviceRoles(
-    state.login.codePageMyDeviceRole,
-    codePageOtherDeviceRole,
-    false
-  )
-  if (!codePageMode) {
-    logger.warn("mode is null!, can't setCodePageMode. Bailing")
-    return
   }
-
-  yield Saga.put(LoginGen.createSetCodePageMode({codePageMode}))
-  yield Saga.put(LoginGen.createSetOtherDeviceCodeState({codePageOtherDeviceRole}))
 }
 
+// TODO entirely change how this works
 function* navBasedOnLoginAndInitialState(): Saga.SagaGenerator<any, any> {
```

Comment:
this function is gross and not how this should work but i didn't want to touch it too much here